sub-groups/sycl-1.2.1/index.md

```diff
@@ -0,0 +1,157 @@
+# Basic Sub group support
+
+This proposal aims to define an interface for using OpenCL 2.2 sub groups in
+SYCL the provisional SYCL 1.2.1 specification, relying on the underlying
+OpenCL implementation supporting the extension `cl_codeplay_basic_subgroups`.
+
+The extension exposes to programmers the ability to identify sub-groups
+on a work-group, count the number of sub-groups available and perform
+a broadcast from one work-item on a sub-group to the rest.
+
+Details of the execution and memory model changes can be found in the
+documentation for the Codeplay's OpenCL vendor extension `cl_codeplay_basic_subgroups` 
+once available.
+
+## Execution model
+
+When this vendor extension is available, the execution model of SYCL 1.2.1
+is extended to also include support for sub-groups of threads inside of a
+work-group.
+Overall, these sub-groups work following the description of the OpenCL 2.2
+sub-groups, with some restrictions:
+
+* The number of sub-groups available for each work-group is determined 
+at compile-time and remains the same during the execution of the SYCL application.
```

Comment:
I'm not as familiar with the details of the SYCL spec, but looking at it now it seems that current use of "compile-time" means when compiling the SYCL program rather than the SYCL runtime calling `clCompileProgram` or similar.  This information can't be known until `clGetKernelSubGroupInfo` (or `clGetKernelSubGroupInfoCODEPLAY`) can be called.

I'm just unclear on the intended meaning of `compile-time`.